community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/MapSupport.scala

```diff
@@ -84,7 +85,10 @@ class LazyMap[T](ctx: QueryContext, ops: Operations[T], id: Long)
   override def size(): Int = allProps.size()
 
   //we need a way forcefully load lazy values
-  def load(): Boolean = allProps != null
+  def load(): MapValue =
```

Comment:
If we really need lazy values, then maybe we should upgrade `load()` to something like `MapValue.materialize()` so all maps had this functionality. In general I'm afraid of any lazy values and the bugs they might cause. This also applies to lazily filtered and mapped values, where the filtering/mapping function might not give the same result at the time of execution when it's lazy.